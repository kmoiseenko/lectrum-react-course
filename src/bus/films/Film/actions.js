// Types
import { types } from './types';

import { api } from './../../../api';

export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    filmFetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());
        try {
            const response = await api.film.fetch(id);

            if (response.status === 200) {
                try {
                    const result = await response.json();

                    dispatch(filmActions.fill(result));
                } catch(error) {
                    console.log(error);
                }
            } else {
                const error = {
                    status: response.status
                };

                dispatch(filmActions.setFetchingError(error));
            }

            dispatch(filmActions.stopFetching());
        } catch(error) {
            console.log(error)
        }
    }
});