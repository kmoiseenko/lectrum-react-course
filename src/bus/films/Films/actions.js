// Types
import { types } from './types';

import { api } from './../../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    filmsFetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());
        try {
            const response = await api.films.fetch();

            if (response.status === 200) {
                try {
                    const { results } = await response.json();

                    dispatch(filmsActions.fill(results));
                } catch(error) {
                    console.log(error);
                }
            } else {
                const error = {
                    status: response.status
                };

                dispatch(filmsActions.setFetchingError(error));
            }

            dispatch(filmsActions.stopFetching());
        } catch(error) {
            console.log(error);
        }
    }
});