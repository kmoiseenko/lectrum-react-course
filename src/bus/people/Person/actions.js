// Types
import { types } from './types';

import { api } from './../../../api';

export const personActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.PERSON_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.PERSON_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.PERSON_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.PERSON_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    personFetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.PERSON_FETCH_ASYNC
        });

        dispatch(personActions.startFetching());
        const response = await api.person.fetch(id);
        
        if (response.status === 200) {
            const result = await response.json();

            dispatch(personActions.fill(result));
        } else {
            const error = {
                status: response.status
            };

            dispatch(personActions.setFetchingError(error));
        }

        dispatch(personActions.stopFetching());
    }
});