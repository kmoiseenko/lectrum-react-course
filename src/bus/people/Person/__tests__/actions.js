import { personActions } from "../actions";
import expectExport from "expect";

describe("personActions", () => {
    test("startFetching", () => {
        expectExport(personActions.startFetching()).toMatchSnapshot();
    });

    test("stopFetching", () => {
        expectExport(personActions.stopFetching()).toMatchSnapshot();
    });

    test("fill", () => {
        expectExport(personActions.fill()).toMatchSnapshot();
    });

    test("setFetchingError", () => {
        expectExport(personActions.setFetchingError()).toMatchSnapshot();
    });

    test("personFetchAsync", () => {
        expectExport(personActions.personFetchAsync()).toMatchSnapshot();
    });
});