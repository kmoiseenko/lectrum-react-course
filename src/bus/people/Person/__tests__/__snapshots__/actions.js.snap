// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`personActions fill 1`] = `
Object {
  "payload": undefined,
  "type": "PERSON_FILL",
}
`;

exports[`personActions personFetchAsync 1`] = `[Function]`;

exports[`personActions setFetchingError 1`] = `
Object {
  "error": true,
  "payload": undefined,
  "type": "PERSON_SET_FETCHING_ERROR",
}
`;

exports[`personActions startFetching 1`] = `
Object {
  "type": "PERSON_START_FETCHING",
}
`;

exports[`personActions stopFetching 1`] = `
Object {
  "type": "PERSON_STOP_FETCHING",
}
`;
